plugins/filter_kubernetes/kube_meta.c

```diff
@@ -87,14 +89,148 @@ static int file_to_buffer(const char *path,
     return 0;
 }
 
+/* Run command to get Kubernetes authorization token */
+static int get_token_with_command(const char *command, 
+                                  char **out_buf, size_t *out_size)
+{
+    FILE *fp;
+    char buf[FLB_KUBE_TOKEN_BUF_SIZE];
+    char *temp;
+    char *res;
+    size_t size = 1;
+    size_t len = 0;
+
+    fp = popen(command, "r");
+    if (fp == NULL) {
+        return -1;
+    }
+
+    res = flb_calloc(1, FLB_KUBE_TOKEN_BUF_SIZE);
+    if (!res) {
+        flb_errno();
+        fclose(fp);
+        return -1;
+    }
+    
+    while (fgets(buf, sizeof(buf), fp) != NULL) {
+        len = strlen(buf);
+        if (len >= FLB_KUBE_TOKEN_BUF_SIZE - 1) {
+            temp = flb_realloc(res, FLB_KUBE_TOKEN_BUF_SIZE * 2);
+            if (temp != NULL) {
+                res = temp;
+            } 
+            else {
+                flb_errno();
+                flb_free(res);
+                fclose(fp);
+                return -1;
+            }
+        }
+        strcpy(res + size - 1, buf);
+        size += len;
+    }
+
+    if (strlen(res) < 1) {
+        flb_free(res);
+        fclose(fp);
+        return -1;
+    }
+    
+    fclose(fp);
+
+    *out_buf = res;
+    *out_size = strlen(res);
+
+    return 0;
+}
+
+/* Set K8s Authorization Token and get HTTP Auth Header */
+static int get_http_auth_header(struct flb_kube *ctx) 
+{
+    int ret;
+    char *temp;
+    char *tk = NULL;
+    size_t tk_size = 0;
+    
+    if (ctx->kube_token_command != NULL) {
+        ret = get_token_with_command(ctx->kube_token_command, &tk, &tk_size);
+        if (ret == -1) {
+            flb_plg_warn(ctx->ins, "failed to run command %s", ctx->kube_token_command);
+        }
+        ctx->kube_token_create = time(NULL);
+    } 
+    else {
+        ret = file_to_buffer(ctx->token_file, &tk, &tk_size);
+        if (ret == -1) {
+            flb_plg_warn(ctx->ins, "cannot open %s", FLB_KUBE_TOKEN);
+        }
+        /* Token from token file will not expire */
+        /* Set the creation time to 0 to aviod refresh */
+        ctx->kube_token_create = 0;
+    }
+
+    /* Token */
+    if (ctx->token != NULL) {
+        flb_free(ctx->token);
+    }
+    ctx->token = tk;
+    ctx->token_len = tk_size;
+
+    flb_plg_info(ctx->ins, "Current Token: %s", ctx->token);
```

Comment:
can we merge these two info into one?

Reply:
It is just for testing. I will remove these two in the final commit. (EKS Fargate team asks for some log info)